runhouse/servers/env_servlet.py

```diff
@@ -297,11 +336,26 @@ def _status_local_helper(self):
 
         # Try loading GPU data (if relevant)
         env_gpu_usage = (
-            self._get_env_gpu_usage(int(env_servlet_pid))
-            if cluster_config.get("has_cuda", False)
-            else {}
+            self._get_env_gpu_usage() if cluster_config.get("has_cuda", False) else {}
+        )
+
+        cluster_config = obj_store.cluster_config
+        interval_size = cluster_config.get(
+            "status_check_interval", DEFAULT_STATUS_CHECK_INTERVAL
         )
 
+        # TODO: [sb]: once introduced, we could use ClusterServlet _cluster_periodic_thread_alive() to replace the
+        #  'should_send_status_and_logs_to_den' logic below.
+        # Only if one of these is true, do we actually need to get the status from each EnvServlet
+        should_send_status_and_logs_to_den: bool = (
+            configs.token is not None and interval_size != -1
+        )
+
+        # rest the gpu_info only if the current env_gpu collection will be sent to den. Otherwise, keep collecting it.
+        if should_send_status_and_logs_to_den:
+            with self.lock:
```

Comment:
To reduce memory usage on the cluster. (The gpu usage we collect is saved on the clusterServlet => once the info is sent to den we can reset the `self.gpu_info`, since the gpu info has been saved in den)